src/seeds/middleware.ts

```diff
@@ -1,25 +1,25 @@
 import { isValidChecksumAddress } from 'ethereumjs-util'
-import { Request, Response, NextFunction } from 'express'
+import { Response, NextFunction } from 'express'
 import Web3 from 'web3';
 
+import { AuthRequest } from './types';
+
 export const authMiddleware = (
-  request: Request,
+  request: AuthRequest,
   response: Response,
   next: NextFunction,
 ) => {
   // skip preflight requests
   if (request.method === 'OPTIONS') { next(); return; }
 
   try {
-    let signer: any;
-
     const auth = {
       message: JSON.stringify(request.body),
       signature: request.header('x-signature') || '',
     }
 
     try {
-      signer = validateSignature(auth, permittedAdminAddresses());
+      request.signer = validateSignature(auth, permittedAdminAddresses());
```

Comment:
In the next PR, can we go ahead and add an API endpoint for the zora approvals so that we can make it public? (I imagine a new route specifically for approving contracts that allows for anyone to submit).

There's a few different ways we can do this, so plz think about it and let me know if you have a better suggestion, but here's one suggested design:
 - Message-based authorization, rather than global authorization
   - validateSignature just recovers and verifies the signature and puts it on the request
   - move middleware.ts#L50 into it's own function called onlyAdmin
   - Rename /seeds/ to /messages/ so that it's a bit more generic
   - Rename validateSeed  -> validateMessage
   - POST /messages/ endpoint does the onlyAdmin check within the validateMessage as part of validation
  - New message type for approving Zora Contracts
    - There is a 1:1 mapping from SeedPayload.CrdtOperation to crdtOperationMessageFnMap, we want to separate this since the Zora approvals will use a new payload operation, but will still be persisted to the DB as a basic update.
    - Create a new message type called CONTRACT_APPROVAL
    - For this operation, the payload should only allow nftFactories and should require exactly NFTFactoryValidUpdateKeys in the payload and should verify that the provided contract is a Zora contract. It can also skip the admin check.

Reply:
Already have the next one in the pipeline! https://github.com/spinamp/spindexer/pull/122

I agree with the new message type for CONTRACT_APPROVAL, and having a per-message authorization that isn't restricted to the admins.